CONTRIBUTING.md

```diff
@@ -1,42 +1,229 @@
+If you're adding a new module or feature, put some thought into where it best
+fits in based on what it does.
+
+```
+  .-----------------.
+  |     openhtf     |
+  |-----------------|
+  | Python package. |
+  '-----------------'
+          |
+          |    .---------------------------------.
+          |    |              conf               |
+          |--->|---------------------------------|
+          |    | Reads and stores configuration. |
+          |    '---------------------------------'
+          |
+          |    .-------------------------.
+          |    |           exe           |
+          |--->|-------------------------|
+          |    | Manages test execution. |
+          |    '-------------------------'
+          |
+          |    .----------------------------------------.
+          |    |                   io                   |
+          |--->|----------------------------------------|
+          |    | Manages UI, logging, and test records. |
+          |    '----------------------------------------'
+          |
+          |    .---------------------------------------------------.
+          |    |                       plugs                       |
+          |--->|---------------------------------------------------|
+          |    | Extensions for interfacing with various hardware. |
+          |    '---------------------------------------------------'
+          |
+          |    .-----------------------------------.
+          |    |               util                |
+          '--->|-----------------------------------|
+               | Utility functions and misc tools. |
+               '-----------------------------------'
+```
+
+
+## Setting Up Your Dev Environment
 The OpenHTF codebase is set up to use pip/virtualenv to manage dependencies.
 
 Pip is the python package manager, while virtualenv is a tool to isolate python
 environments. You'll need both in order to work with the OpenHTF codebase.
 
-Pip installs requirements (dependencies) from pypi (the python package index).
-It can use a requirements.txt file to know which packages to install. Typically
-this file points at hosted package names in pypi to install but it can also
-install local packages in development mode (using the -e flag).  This causes pip
+Pip installs dependencies from pypi (the python package index), but it can also
+install local packages in development mode (using the -e flag). This causes pip
 to install the dependency using symlinks from the filesystem so changes made to
 those files are instantly reflected.
 
-By design, virtualenv and OpenHTF's python dependencies are not included in the
+Neither virtualenv nor OpenHTF's python dependencies are included in the
 repository. The following steps set up a new virtualenv environment and install
-OpenHTF's dependencies into it using pip (steps 5-7 below).
+OpenHTF's dependencies into it using pip.
+
+
+### Linux
```

Comment:
this is really only debian derivatives because of the usage of `apt`, at some point we might go through the exercise of installing on red hat and potentially a couple other flavors, but for now it's worth noting that these instructions are for debian-based systems (including ubuntu).

as part of PyPI launch, we should probably include a list of OS'es and versions that we have explicitly verified support for.